spec/datadog/core/environment/cgroup_spec.rb

```diff
@@ -132,6 +132,26 @@
           is_expected.to include(be_a_kind_of(described_class::Descriptor))
         end
       end
+
+      context 'in a Fargate 1.4+ (2-part short random) environment' do
+        include_context 'Fargate 1.4+ (2-part short random) environment'
+
+        it do
+          is_expected.to be_a_kind_of(Array)
+          is_expected.to have(11).items
+          is_expected.to include(be_a_kind_of(described_class::Descriptor))
+        end
+      end
+
+      context 'in a Fargate 1.4+ with ECS+docker environment' do
+        include_context 'Fargate 1.4+ with ECS+docker environment'
+
+        it do
+          is_expected.to be_a_kind_of(Array)
+          is_expected.to have(10).items
```

Comment:
Minor: I assume this array has a number of entries that matches the number of entries on the `cgroup_file`. If that's the case, perhaps we could define that as a `let` in the shared context, so that here we always test "expect to have as many entries as was put in the shared context", as otherwise these numbers seem to just "appear" here.

Bonus points: we could even factor out the "it" in that case, which is something I don't often advocate, but I think in this case it kinda makes sense to write it as `include_examples 'parsing cgroup file into an array of descriptors`.

Reply:
Refactored as suggested, good call!